tests/org.eclipse.n4js.flowgraphs.tests/model/Destructuring.n4js.xt

```diff
@@ -0,0 +1,334 @@
+	// Destructuring5 XPECT succs at '{x,y} = {x:1, y:[2]}' --> let {x\,y} = {x:1\, y:[2]};
+	// Destructuring5 XPECT succs at 'let {x,y} = {x:1, y:[2]};' --> "end"
+	"start";
+	let {x,y} = {x:1, y:[2]};
+	"end";
+}
+
+function Destructuring6() {
+	// Destructuring6 XPECT container --> FunctionDeclaration::Block
+	/* Destructuring6 XPECT allEdges ---
+		"start" --> x,
+		x --> y,
+		y --> 1,
+		1 --> 2,
+		2 --> {y:2},
+		{y:2} --> [1\, {y:2}],
+		[1\, {y:2}] --> [x\,y] = [1\, {y:2}],
+		[x\,y] = [1\, {y:2}] --> let [x\,y] = [1\, {y:2}];,
+		let [x\,y] = [1\, {y:2}]; --> "end"
+	--- */
+	/* Destructuring6 XPECT allPaths ---
+	   "start" -> x -> y -> 1 -> 2
+	   -> {y:2}
+	   -> [1\, {y:2}]
+	   -> [x\,y] = [1\, {y:2}]
+	   -> let [x\,y] = [1\, {y:2}];
+	   -> "end"
+	--- */
+	// Destructuring6 XPECT path from '"start"' to '"end"' -->
+	// Destructuring6 XPECT commonPreds of '1' and '2' --> 1
+	// Destructuring6 XPECT succs at '"start"' --> x
+	// Destructuring6 XPECT succs at 'x' --> y
+	// Destructuring6 XPECT succs at 'y' --> 1
+	// Destructuring6 XPECT succs at '1' --> 2
+	// Destructuring6 XPECT succs at '2' --> {y:2}
+	// Destructuring6 XPECT succs at '{y:2}' --> [1\, {y:2}]
+	// Destructuring6 XPECT succs at '[1, {y:2}]' --> [x\,y] = [1\, {y:2}]
+	// Destructuring6 XPECT succs at '[x,y] = [1, {y:2}]' --> let [x\,y] = [1\, {y:2}];
+	// Destructuring6 XPECT succs at 'let [x,y] = [1, {y:2}];' --> "end"
+	"start";
+	let [x,y] = [1, {y:2}];
+	"end";
+}
+
+function Destructuring7() {
+	// Destructuring7 XPECT container --> FunctionDeclaration::Block
+	// Destructuring7 XPECT allIslandElems -->
+	// Destructuring7 XPECT allCatchBlocks -->
+	/* Destructuring7 XPECT allEdges ---
+		"start" --> x,
+		x --> 1,
+		1 --> {y : 1},
+		{y : 1} --> null,
+		null --> [ {x} = {y : 1} ] = null,
+		[ {x} = {y : 1} ] = null --> var [ {x} = {y : 1} ] = null;,
+		var [ {x} = {y : 1} ] = null; --> "end"
+	--- */
+	/* Destructuring7 XPECT allPaths ---
+		"start" -> x -> 1 -> {y : 1} -> null -> [ {x} = {y : 1} ] = null -> var [ {x} = {y : 1} ] = null; -> "end"
+	--- */
+	/* Destructuring7 XPECT allPaths direction 'Backward' ---
+		"end" -> var [ {x} = {y : 1} ] = null; -> [ {x} = {y : 1} ] = null -> null -> {y : 1} -> 1 -> x -> "start"
+	--- */
+	// Destructuring7 XPECT path from '"start"' to '"end"' -->
+	// Destructuring7 XPECT commonPreds of 'null' and 'x' --> x
+	// Destructuring7 XPECT succs at '"start"' --> x
+	// Destructuring7 XPECT succs at 'x' --> 1
+	// Destructuring7 XPECT succs at '1' --> {y : 1}
+	// Destructuring7 XPECT succs at '{y : 1}' --> null
+	// Destructuring7 XPECT succs at 'null' --> [ {x} = {y : 1} ] = null
+	// Destructuring7 XPECT succs at '[ {x} = {y : 1} ] = null' --> var [ {x} = {y : 1} ] = null;
+	// Destructuring7 XPECT succs at 'var [ {x} = {y : 1} ] = null' --> "end"
+	"start";
+	var [ {x} = {y : 1} ] = null;
+	"end";
+}
+
+
+function Destructuring8(q) {
+	// Destructuring7 XPECT container --> FunctionDeclaration::Block
```

Comment:
wrong index (should be `// Destructuring8 XPECT ...`)

Reply:
thanks